MathLib/TemplatePoint.h

```diff
@@ -92,6 +92,21 @@ template <typename T, std::size_t DIM = 3> class TemplatePoint
 		std::copy(std::istream_iterator<T>(is), std::istream_iterator<T>(), _x.begin());
 	}
 
+	/// operator -=
+	inline void operator-= (const TemplatePoint<T,DIM> &p)
```

Comment:
I modified the branch a little bit. If you want you can take a look [here](https://github.com/TomFischer/ogs/commits/lower-dim).